Github actions updated XCode, automated build broken on MacOS

```
 [ 47%] Built target rsa_discovery_etcd_bundle
/Users/runner/work/celix/celix/bundles/pubsub/pubsub_admin_tcp/src/pubsub_tcp_handler.c:57:9: error: 'MSG_NOSIGNAL' macro redefined [-Werror,-Wmacro-redefined]
#define MSG_NOSIGNAL (0)
        ^
/Applications/Xcode_12.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.0.sdk/usr/include/sys/socket.h:584:9: note: previous definition is here
#define MSG_NOSIGNAL    0x80000         /* do not generate SIGPIPE on EOF */
```

--- a/bundles/pubsub/pubsub_admin_tcp/src/pubsub_tcp_handler.c
+++ b/bundles/pubsub/pubsub_admin_tcp/src/pubsub_tcp_handler.c
@@ -54,7 +54,9 @@
 #define MAX_DEFAULT_BUFFER_SIZE 4u
 
 #if defined(__APPLE__)
+#ifndef MSG_NOSIGNAL
 #define MSG_NOSIGNAL (0)
+#endif
 #endif
 
 #define L_DEBUG(...) \

